packages/headless/src/user/provider.tsx

```diff
@@ -21,6 +23,16 @@ const cache = new LocalStorage();
 
 const userCache = cache.get<Data>(CacheKey.User);
 
+// routes that can be visited without auth
+const noAuthRoutes: string[] = [
+  Web.Login,
+  Web.Register,
+  Web.ResetPassword,
+  Web.ForgetPassword,
+  Web.VerifyEmail,
+  Web.CompleteProfile,
+];
+
```

Comment:
This provider is used in the web, dashboard, mobile (in the future). This logic should be moved to `web/src/pages/Root.tsx`. Also, we should implement something similar to in the dashboard.